### PR TITLE
[TR] Remove step I should see * flash message

### PR DESCRIPTION
**Pull request by willy-ahva**
Done with @skeleton :)

This PR : 
 - removed the use of flash messages.
 - remove 4 waits.
 - message argument for spin is mandatory.
 - add messages on all spins.
 - add stability.
 - add new 0 failing behats.
 - fix 7 failing behats.
 - 0 PHP Fatal error

What we did :

 - We removed duplicated methods `I should see flash message blabla` and `I should see a flash message blabla` to use only `I should see the flash message blabla`.

 - We removed the content of the method `IShouldSeeTheFlashMessage` to always return true. We kept this step to be able to re-use it where we have to. When we will have flash messages that not disappear, we will just have to implement the method.

When we removed implementation of flash messages we encountered some problems.

 - Inside the method there was a spin which allowed us to wait to see the flash message. Remove it has some collateral damages. Steps which are after stopped to pass because it was too fast and they have no time to see what they need.

 - Fixing steps that were broken implies that we have to removed some important waits. One in `iVisitTheTab`, `iVisitTheGroup`, `iChangeTheTo()`, and one in `IPressTheButton` => sh*t explosion.

 - The `IPressTheButton` wait was pretty hard to fix. Each time we save a page, the save process take from 2 to 4 or more seconds. As we stopped to wait, we pressed the save button then execute directly the next step, so we were never able to save a product, a user, a variant group, etc. It broke a lot of scenarii.

 - I tried to fix it spinning on each next step the element i should see. Unfortunately, when we save we continue to see the page in background, so spins were always able to found their element before the saving process was finished.

 - The only way I found to fix it is to add a step after the save : `I should not see the text "There are unsaved changes."` to be able to wait the end of saving process and continue to the next one.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | /
| Added Behats                      | /
| Changelog updated                 | no
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: features/Context/AssertionContext.php [hunk 1]
```diff
@@ -257,6 +243,8 @@ public function iShouldSeeFlashMessage($text)
      */
     public function iShouldSeeTheFlashMessage($text)
     {
+        return true;
+
```
Comment (damien-carcel) [inline features/Context/AssertionContext.php]: Why do you keep the code below if you do a "return true" here?

File: features/Context/AssertionContext.php [hunk 2]
```diff
@@ -235,28 +235,16 @@ public function theFieldsShouldBeDisabled($fields)
...
-        return true;
-    }
-
-    /**
      * @param $text
      *
-     * @Then /^I should see the flash message "([^"]*)"$/
+     * @Then /^I should see the flash message "(.*)"$/
      *
      * @throws Spin\TimeoutException
```
Comment (pierallard) [inline features/Context/AssertionContext.php]: add @deprecated please

File: features/Context/Page/Base/Base.php [hunk 3]
```diff
@@ -270,7 +269,7 @@ protected function getConfirmDialog()
     {
         $element = $this->getElement('Dialog');
 
-        if (!$element) {
+        if (null === $element) {
```
Comment (jjanvier) [inline features/Context/Page/Base/Base.php]: great Lord, you saved so many kittens with that :+1:

File: features/Context/Page/Base/Form.php [hunk 4]
```diff
@@ -430,7 +439,9 @@ public function fillField($field, $value, Element $element = null)
      */
     public function getHistoryRows()
     {
-        return $this->getElement('Updates grid')->findAll('css', 'tbody tr');
+        return $this->spin(function () {
+            return $this->getElement('Updates grid')->findAll('css', 'tbody tr');
```
Comment (juliensnz) [inline features/Context/Page/Base/Form.php]: History grid ?

**Comment by jjanvier** [comment]
Really good :+1: 

Apart from the `I should not see the text "There are unsaved changes."` GTM.

But `I should not see the text "There are unsaved changes."` is a really serious problem. If @juliensnz has no idea how to fix it then it's GTM. But it's really as a last solution. And we should fix this asap. But as this PR is big step forward, I'm OK if we merge like this.

File: features/Context/Page/Base/Form.php (hunk 4 above)
Comment (willy-ahva) [inline features/Context/Page/Base/Form.php]: Pls add better comments

**Comment by juliensnz** [comment]
GTM

**Pull request merged**